src/main/java/org/ossgang/commons/monads/Maybe.java

```diff
@@ -278,6 +278,22 @@ public void throwOnException() {
         return attempt(() -> function.apply(value));
     }
 
+    /**
+     * Apply a transformation function if this {@link Maybe} is in a "unsuccessful" state. Pass through the value in
+     * case it is in an "successful" state. If the transformation function throws, the exception is returned wrapped
+     * in an "unsuccessful" Maybe.
+     *
+     * @param function the function to apply
+     * @return An unsuccessful Maybe the new exception if the transformation succeeds. An unsuccessful Maybe otherwise.
+     */
+    public Maybe<T> mapException(ThrowingFunction<Throwable, Throwable> function) {
+        requireNonNull(function);
+        if (exception == null) {
+            return Maybe.ofValue(value);
```

Comment:
return this? Just in case we ever have subclasses ...

Reply:
OK, I see, the map() etc. do the same ... no strong preference.

Reply:
that's actually a problem in case of Maybe<Void> ! Since we cannot create a `Maybe.ofValue(null) ` :sweat_smile:
Usual pitfall.. I fixed also the recover method